diff --git a/src/ads_globals.h b/src/ads_globals.h
index 53e612b..93c7360 100644
--- a/src/ads_globals.h
+++ b/src/ads_globals.h
@@ -30,8 +30,10 @@
 //============================================================================
 //                                   INCLUDES
 //============================================================================
+#include <QtGlobal>
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
 #include <QPair>
-#include <QtCore/QtGlobal>
+#endif
 #include <QPixmap>
 #include <QWidget>
 #include <QDebug>
@@ -207,6 +209,7 @@ void replaceSplitterWidget(QSplitter* Splitter, QWidget* From, QWidget* To);
 void hideEmptyParentSplitters(CDockSplitter* FirstParentSplitter);
 
 
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
 /**
  * Convenience class for QPair to provide better naming than first and
  * second
@@ -219,6 +222,16 @@ public:
 	bool append() const {return this->second;}
 	int insertOffset() const {return append() ? 1 : 0;}
 };
+#else
+class CDockInsertParam : public std::pair<Qt::Orientation, bool>
+{
+public:
+    using std::pair<Qt::Orientation, bool>::pair;
+	Qt::Orientation orientation() const {return this->first;}
+	bool append() const {return this->second;}
+	int insertOffset() const {return append() ? 1 : 0;}
+};
+#endif
 
 /**
  * Returns the insertion parameters for the given dock area
